internal/host/validator.go

```diff
@@ -658,4 +660,172 @@ func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationC
+						if ip == ipv6 {
+							return h.RequestedHostname, h.Role, nil
+						}
+					}
+				}
+			}
+		}
+	}
+	return "", "", fmt.Errorf("host with IP %s not found in inventory", ip)
+}
+
+func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host) (map[string]string, error) {
+	var connectivityReport models.ConnectivityReport
+	err := json.Unmarshal([]byte(host.Connectivity), &connectivityReport)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshall host connectivity for %s:%s", host.ID, err)
+	}
+	latency := map[string]string{}
+	for _, r := range connectivityReport.RemoteHosts {
+		for _, l3 := range r.L3Connectivity {
+			if l3.AverageRTTMs > clusterRoleReqs.Ocp.NetworkLatencyThresholdMs {
+				if _, ok := latency[l3.RemoteIPAddress]; !ok {
+					hostname, role, err := getHostnameAndRoleByIP(l3.RemoteIPAddress, hosts)
+					if err != nil {
+						return nil, err
+					}
+					if role == host.Role {
+						latency[l3.RemoteIPAddress] = hostname
+					}
+				}
+			}
+		}
+	}
+	return latency, nil
+}
+
+func (v *validator) printHasNetworkLatencyRequirementForRole(c *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return "Network latency requirement have been satisfied."
+	case ValidationFailure:
+		hosts, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+		if hosts == nil && err != nil {
+			return fmt.Sprintf("Error while attempting to validate network latency: %s", err)
+		}
+		s := strings.Builder{}
+		s.WriteString(fmt.Sprintf("Network latency requirements of less or equals than %.3f ms not met for connectivity between %s and ", c.clusterHostRequirements.Ocp.NetworkLatencyThresholdMs, c.host.RequestedHostname))
+		var keys []string
+		for _, k := range reflect.ValueOf(hosts).MapKeys() {
+			keys = append(keys, k.String())
+		}
+		sort.Strings(keys)
+		for i := 0; i < len(keys)-1; i++ {
+			s.WriteString(hosts[keys[i]] + ", ")
+		}
+		if len(keys) > 0 {
+			s.WriteString(hosts[keys[len(keys)-1]])
+		}
+		s.WriteString(".")
+		return s.String()
+	case ValidationPending:
+		return "Missing network latency information."
+	default:
+		return fmt.Sprintf("Unexpected status %s", status)
+	}
+}
+
+func (v *validator) hasPacketLossRequirementForRole(c *validationContext) ValidationStatus {
+
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending
+	}
+	hosts, err := v.validatePacketLossForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+	if err != nil || len(hosts) > 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func (v *validator) validatePacketLossForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host) (map[string]string, error) {
+	var connectivityReport models.ConnectivityReport
+	err := json.Unmarshal([]byte(host.Connectivity), &connectivityReport)
```

Comment:
That code appears also in https://github.com/openshift/assisted-service/pull/1432/files?file-filters%5B%5D=.go#diff-d7db3d1eb3c67167cb6aaffa4ed66cd1dcd7d2097d63648ecd03410a36f1283eR711
please extract a common function (maybe in `hostutil`) package

Reply:
Got it.